### PR TITLE
Demo examples

### PR DESCRIPTION
**Pull request by DenisRybas**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -7,7 +7,7 @@ name = 'pack_encrypted'
 harness = false
 
 [[example]]
-name = 'basic'
+name = 'advanced_params'
```
Comment (vimmerru) [inline Cargo.toml:10]: I would expect all examples must be in Cargo.toml

File: examples/multi_recipient.rs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    .from(ALICE_DID.to_owned())
+    .finalize();
+
+    // --- Packing encrypted and authenticated message for Bob ---
+    let did_resolver = ExampleDIDResolver::new(vec![
+        ALICE_DID_DOC.clone(),
+        BOB_DID_DOC.clone(),
+        CHARLIE_DID_DOC.clone(),
+    ]);
+    let secrets_resolver = ExampleSecretsResolver::new(ALICE_SECRETS.clone());
```
Comment (vimmerru) [inline examples/multi_recipient.rs:37]: I suggest keep empty line after multi-line operator

File: examples/multi_recipient.rs [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+                forward: false, // Forward wrapping is unsupported in current version
+                ..PackEncryptedOptions::default()
+            },
+        )
+        .await
+        .expect("Unable pack_encrypted");
+
+    // --- Sending message to Charlie ---
+    println!("Sending message to Charlie \n{}\n", msg_charlie);
+    println!(
```
Comment (vimmerru) [inline examples/multi_recipient.rs:76]: I suggest keep empty line before multi-line operator

File: examples/multi_recipient.rs [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    .await
+    .expect("Unable unpack");
+
+    println!("Bob received message is \n{:?}\n", msg);
+    println!("Bob received message unpack metadata is \n{:?}\n", metadata);
+
+    // --- Unpacking message for Charlie ---
+    let did_resolver =
+        ExampleDIDResolver::new(vec![ALICE_DID_DOC.clone(), CHARLIE_DID_DOC.clone()]);
+    let secrets_resolver = ExampleSecretsResolver::new(CHARLIE_SECRETS.clone());
```
Comment (vimmerru) [inline examples/multi_recipient.rs:102]: I suggest keep empty line after multi-line operator

File: examples/multi_recipient.rs [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        &secrets_resolver,
+        &UnpackOptions {
+            ..UnpackOptions::default()
+        },
+    )
+    .await
+    .expect("Unable unpack");
+
+    println!("Charlie received message is \n{:?}\n", msg);
+    println!(
```
Comment (vimmerru) [inline examples/multi_recipient.rs:116]: I suggest keep empty line before multi-line operator

File: src/test_vectors/did_doc/mod.rs [hunk 6]
```diff
@@ -1,6 +1,8 @@
 mod alice;
 mod bob;
 mod charlie;
+mod mediator1;
+mod mediator2;
```
Comment (vimmerru) [inline src/test_vectors/did_doc/mod.rs:5]: You should also re-export as

```
// TODO: Remove allow
#[allow(unused_imports)]
pub(crate) use mediator1::*;
```

File: src/test_vectors/secrets/mod.rs [hunk 7]
```diff
@@ -2,6 +2,8 @@ mod alice;
 mod bob;
 mod charlie;
 mod charlie_rotated_to_alice;
+mod mediator1;
```
Comment (vimmerru) [inline src/test_vectors/secrets/mod.rs:5]: You should also re-export as

```
// TODO: Remove allow
#[allow(unused_imports)]
pub(crate) use mediator1::*;
```

File: examples/repudiable_authenticated_encryption.rs [hunk 8]
```diff
@@ -15,19 +15,17 @@ use test_vectors::{ALICE_DID, ALICE_DID_DOC, ALICE_SECRETS, BOB_DID, BOB_DID_DOC
```
Comment (vimmerru) [inline examples/repudiable_authenticated_encryption.rs:15]: I would suggest to keep `basic` name for the test. And provide same tests as python version

image

**Review by vimmerru — CHANGES_REQUESTED** [review]
Please check my comments

File: Cargo.toml (hunk 1 above)
Comment (DenisRybas) [reply Cargo.toml:13]: fixed

File: examples/multi_recipient.rs (hunk 2 above)
Comment (DenisRybas) [reply examples/multi_recipient.rs]: fixed

File: examples/multi_recipient.rs (hunk 3 above)
Comment (DenisRybas) [reply examples/multi_recipient.rs]: fixed

File: examples/multi_recipient.rs (hunk 4 above)
Comment (DenisRybas) [reply examples/multi_recipient.rs]: fixed

File: examples/multi_recipient.rs (hunk 5 above)
Comment (DenisRybas) [reply examples/multi_recipient.rs]: fixed

File: src/test_vectors/secrets/mod.rs (hunk 7 above)
Comment (DenisRybas) [reply src/test_vectors/secrets/mod.rs:5]: fixed

File: examples/repudiable_authenticated_encryption.rs (hunk 8 above)
Comment (DenisRybas) [reply examples/repudiable_authenticated_encryption.rs]: all tests except mediators provided

**Review by vimmerru — APPROVED** [review]
LGTM

**Pull request merged**